kiwi/bootloader/config/grub2.py

```diff
@@ -1459,19 +1459,24 @@ def _fix_grub_loader_entries_linux_and_initrd_paths(self):
         )
         for menu_entry_file in glob.iglob(loader_entries_pattern):
             with open(menu_entry_file) as grub_menu_entry_file:
-                menu_entry = grub_menu_entry_file.read()
-                if self.bootpartition:
-                    menu_entry = re.sub(
-                        r'(linux|initrd) .*/boot(.*)',
-                        r'\1 \2',
-                        menu_entry
-                    )
-                else:
-                    menu_entry = re.sub(
-                        r'(linux|initrd) .*(/boot.*)',
-                        r'\1 \2',
-                        menu_entry
-                    )
+                menu_entry = grub_menu_entry_file.read().split(os.linesep)
+
+            for line_number, menu_entry_line in enumerate(menu_entry):
+                if bool(re.match(r'^(linux|initrd) .*', menu_entry_line)):
+
+                    log.debug(f'Existing loader entry: {menu_entry_line}')
+                    config_path = menu_entry_line.split(" ", 1)[0]
```

Comment:
single quotes please ```' '``` instead of ```" "```. Reason here is also consistency and that double quotes for strings in python are typically used when variables or other types of substitution is part of the string expression. As this is not the case here I recommend the single quotes. Sorry if this is nitpick but the more consistent we are the more robust the code will grow. Thanks

After that it looks good to me. Thanks much for your patience :)